pdm/models/repositories.py

```diff
@@ -124,6 +127,35 @@ def find_candidates(
                 key=lambda c: c.version,
                 reverse=True,
             )
+
+            if not sorted_cans:
+                termui.logger.debug(
+                    "\tCould not find any matching candidates even when considering "
+                    "pre-releases.",
+                )
+
+        def print_candidates(
+            title: str, candidates: List[Candidate], max_lines: int = 10
+        ) -> None:
+            termui.logger.debug("\t" + title)
+            logged_lines = set()
+            for can in candidates:
+                new_line = f"\t  {can}"
+                if new_line not in logged_lines:
+                    logged_lines.add(new_line)
+                    if len(logged_lines) > max_lines:
+                        termui.logger.debug(
+                            f"\t  ... [{len(candidates)-max_lines} more candidate(s)]"
+                        )
+                        break
+                    else:
+                        termui.logger.debug(new_line)
```

Comment:
Lack of considering the `candidates` parameter is an empty list. Please inform that in log file.

Reply:
Good catch. Fixed.
Only print list of `cans` if non-empty. If it is empty, the user gets a `pdm.exceptions.CandidateNotFound` anyway.